File: PathoFroh/src/main/java/com/patho/main/repository/jpa/custom/StainingPrototypeRepositoryCustom.java
package com.patho.main.repository.jpa.custom;

import com.patho.main.model.preset.StainingPrototype;
import com.patho.main.model.preset.StainingPrototypeType;

import java.util.List;
import java.util.Optional;

public interface StainingPrototypeRepositoryCustom {

    /**
     * Returns a patient with the given id
     *
     * @param id
     * @return
     */
    Optional<StainingPrototype> findOptionalByIdAndInitilize(Long id, boolean initializeBatch);

    /**
     * @return
     */
    List<StainingPrototype> findAllByTypeIgnoreArchivedOrderByPriorityCountDesc(StainingPrototypeType type,
                                                                                boolean initializeBatch, boolean irgnoreArchived);
}

File: PathoFroh/src/main/kotlin/com/patho/main/repository/jpa/custom/StainingPrototypeRepositoryCustom.kt
package com.patho.main.repository.jpa.custom

import com.patho.main.model.preset.StainingPrototype
import com.patho.main.model.preset.StainingPrototypeType
import java.util.*

interface StainingPrototypeRepositoryCustom {

    fun findOptionalByIdAndInitialize(id: Long): Optional<StainingPrototype>

    fun findAllByTypeIgnoreArchivedOrderByPriorityCountDesc(type: StainingPrototypeType,
                                                            ignoreArchived: Boolean): List<StainingPrototype>
}